### ISSUE TITLE
AKS 1.19.X missing metrics & alerts in Azure Monitor for containers

### ISSUE THREAD

**Issue by saaror** [Azure/AKS#1999]
**What happened**:
Starting AKS version 1.19.X due to k8s upstream issue, Azure Monitor for containers customers will not see all the metrics under 'insights.container/nodes'  metrics namespace & the recommended alerts using these metrics will be broken.

This is due to the k8s upstream issue of [missing cAdvisor metrics](https://github.com/kubernetes/kubernetes/issues/95204)
**Anything else we need to know?**:
This issue will be fixed once the k8s upstream issue is fixed & backported to AKS.

Please reach out to akscoin@microsoft for any question.
**Environment**:
AKS version 1.19.X.

**Comment by saaror** [comment]
Impact: High, breaking existing customer alerts.

**Comment by vishiy** [comment]
Caused by a kubernetes bug that is tracked by - https://github.com/kubernetes/kubernetes/issues/95204

**Comment by wesselkranenborg** [comment]
This bug is now fixed upstream, is there any ETA on when it will be released on AKS?

**Comment by kasbst** [comment]
It is possible to handle this with some urgency now? Node metrics are critical part of AKS monitoring... Thanks!

**Comment by saaror** [comment]
Thanks folks for the patience. The fix is completed on AKS side & should be rolling out AKS next week release in 1.19.7 version.

**Comment by kasbst** [comment]
@saaror Thanks! But when I check available upgrades for eastus2 there is only 1.19.6 version. Does this version contain the fix as well?

**Comment by adhodgson1** [comment]
I was receiving the following error messages when attempting to deploy the 2 metrics memoryWorkingSetPercentage and cpuUsagePercentage. I tried running the deployment after upgrading to AKS 1.19.7 in NorthEurope but continue to get this error. Am I missing something else?
```
"Message": "The following metric name(s) were not found: memoryWorkingSetPercentage. Please note that for custom metrics, the relevant metric namespace must be specified. Activity ID: b88057d0-1658-4060-bb36-b1ed8a4e2b2b."
"Message": "The following metric name(s) were not found: cpuUsagePercentage. Please note that for custom metrics, the relevant metric namespace must be specified. Activity ID: 74a35ec4-e3d9-446e-9730-41c0992ea43e."
```

**Comment by aristosvo** [comment]
Have you activated the [Monitoring Metrics Publisher role](https://docs.microsoft.com/en-us/azure/azure-monitor/insights/container-insights-update-metrics) for your cluster?

**Comment by adhodgson1** [comment]
@aristosvo I am creating the relevant RBAC entry as part of the deployment as the account that runs the ARM template doesn't have the capability to update RBAC.  Here is the Ansible code:

```
- name: Ensure AKS Monitoring identity has 'Monitoring Metrics Publisher' rights to the AKS resource
  azure_rm_roleassignment:
    scope: "{{ aks_resource_id }}"
    assignee_object_id: "{{ oms_managed_identity_object_id }}"
    role_definition_id: "/subscriptions/{{ azure_subscription_id }}/providers/Microsoft.Authorization/roleDefinitions/3913510d-42f4-4e42-8a64-420c390055eb"
```
I can share the ARM template if required.

**Comment by aristosvo** [comment]
@adhodgson1 the setup is the same as I have, it seems still broken in 1.19.7 😥

**Comment by tankovich** [comment]
> @adhodgson1 the setup is the same as I have, it seems still broken in 1.19.7 😥

Same here, we just rolled a new 1.19.7 cluster and cannot see all metrics under `insights.container/nodes`.

**Comment by vishiy** [comment]
The fix for the monitoring addon for this issue is in the works. Once the addon is patched, >=1.19.7 & >=1.20.2 clusters will start to collect these metrics.

**Comment by aristosvo** [comment]
@vishiy Thanks for letting us know!

Is there a place where RCAs and mitigation of known issues like these are published, to inform my clients and manager about these kind of things?

**Comment by vishiy** [comment]
@aristosvo - Can you please share more on #2078 ?

**Comment by aristosvo** [comment]
Issue seems resolved in my clusters, thanks!!

**Comment by kasbst** [comment]
I can confirm this is now resolved for us as well - cluster version 1.19.7

**Comment by saaror** [comment]
Closing issue, the fix has been rolled out.
